### ISSUE TITLE
Recent Xcode regression

### ISSUE THREAD

**Issue by traversaro** [robotology/ycm#363]
I suspect there was a regression due either to a CMake update or a Xcode update:
~~~
-- Generating done
CMake Error in 3rdparty/CMakeLists.txt:
  The custom command generating

    /Users/runner/work/robotology-superbuild/robotology-superbuild/build/robotology/YCM/3rdparty/cmrc/CMakeRC.cmake

  is attached to multiple targets:

    3rdparty-cmrc
    3rdparty-catch2

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".


CMake Error in docs/CMakeLists.txt:
  The custom command generating

    /Users/runner/work/robotology-superbuild/robotology-superbuild/build/robotology/YCM/3rdparty/cmrc/CMakeRC.cmake

  is attached to multiple targets:

    3rdparty-cmrc
    3rdparty-catch2

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".
~~~

**Comment by traversaro** [comment]
Related robotology-superbuild issues: 
* https://github.com/robotology/robotology-superbuild/issues/525
* https://github.com/robotology/robotology-superbuild/pull/524

**Comment by drdanz** [comment]
Fixed in #364
